store/store.go

```diff
@@ -296,7 +296,7 @@ type ThreadStore interface {
 	GetMembershipsForUser(userId, teamID string) ([]*model.ThreadMembership, error)
 	GetMembershipForUser(userId, postID string) (*model.ThreadMembership, error)
 	DeleteMembershipForUser(userId, postID string) error
-	MaintainMembership(userID, postID string, following, incrementMentions, updateFollowing, updateViewedTimestamp, updateParticipants bool) (*model.ThreadMembership, error)
+	MaintainMembership(ctx context.Context, userID, postID string, opts ThreadMembershipOpts) (*model.ThreadMembership, error)
```

Comment:
It would be great to have a comment on the interface or implementation for `MaintainMembership` to explain it's purpose (I know it's pre-existing).

Reply:
I remember asking to add a comment inside because the logic was getting a bit complex. I'll try to see if we can expand a bit on that although the team owning this feature may have more context.